### ISSUE TITLE
Parser raises TypeError: unsupported operand type(s) when handling calendar.IllegalMonthError

### ISSUE THREAD

**Issue by o-felixz** [dateutil/dateutil#1038]
The string `æ\x9få\x9b\x9b, 22 å\n«æ\x9c 2013 11:27:40 +0800` is interpreted to have month number 22 - which is rightly rejected by `calendar`. There appears to be a problem with how this exception is handled by `dateutil` however as instead of the expected `ValueError` a `TypeError` is raised instead:

**How to reproduce**

```
>>> parser.parse("æ\x9få\x9b\x9b, 22 å\n«æ\x9c 2013 11:27:40 +0800", fuzzy=True)
```

**Expected**

Some sort of `ValueError`

**Actual**

```
Traceback (most recent call last):
  File "<>/venv/lib/python3.7/site-packages/dateutil/parser/_parser.py", line 655, in parse
    ret = self._build_naive(res, default)
  File "<>/venv/lib/python3.7/site-packages/dateutil/parser/_parser.py", line 1238, in _build_naive
    if cday > monthrange(cyear, cmonth)[1]:
  File "/usr/lib/python3.7/calendar.py", line 124, in monthrange
    raise IllegalMonthError(month)
calendar.IllegalMonthError: bad month number 22; must be 1-12

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<>/venv/lib/python3.7/site-packages/dateutil/parser/_parser.py", line 1374, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "<>/venv/lib/python3.7/site-packages/dateutil/parser/_parser.py", line 657, in parse
    six.raise_from(ParserError(e.args[0] + ": %s", timestr), e)
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

**Environment**
Platform: Ubuntu 18.04.4/Debian 10 (docker container)
Python version: 3.7.5
dateutil version: 2.8.1

**Comment by ffe4** [comment]
The `TypeError` has actually already been fixed on master.

```python
>>> import dateutil
>>> dateutil.__version__
'2.8.2.dev20+gfa8e304.d20200405'
>>> from dateutil import parser
>>> parser.parse("æ\x9få\x9b\x9b, 22 å\n«æ\x9c 2013 11:27:40 +0800", fuzzy=True)
Traceback (most recent call last):
  File "/Users/Daniel/ffe4/dateutil/dateutil/parser/_parser.py", line 655, in parse
    ret = self._build_naive(res, default)
  File "/Users/Daniel/ffe4/dateutil/dateutil/parser/_parser.py", line 1238, in _build_naive
    if cday > monthrange(cyear, cmonth)[1]:
  File "/Users/Daniel/.pyenv/versions/3.8.2/lib/python3.8/calendar.py", line 124, in monthrange
    raise IllegalMonthError(month)
calendar.IllegalMonthError: bad month number 22; must be 1-12

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Daniel/ffe4/dateutil/dateutil/parser/_parser.py", line 1374, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "/Users/Daniel/ffe4/dateutil/dateutil/parser/_parser.py", line 657, in parse
    six.raise_from(ParserError(str(e) + ": %s", timestr), e)
  File "<string>", line 3, in raise_from
dateutil.parser._parser.ParserError: bad month number 22; must be 1-12: æå, 22 å
«æ 2013 11:27:40 +0800
```

Since 2.8.1 you will get a `ParserError`, which is a subclass of ValueError. Documentation of that fact was missed in the stable release, but is available in the latest [online documentation](https://dateutil.readthedocs.io/en/latest/parser.html#dateutil.parser.parse).

**Comment by Pitrified** [comment]
Hello! I don't know if I should open a new issues, but this still happens with the following string `Liverpool NY 13088 42 cases`, as shown in [this](https://stackoverflow.com/questions/66856369/illegalmontherror-in-python-datefinder) SO question. This issue was linked in the comments, so I thought I'd let you know.
With the following version:
```
datefinder             0.7.1
python-dateutil        2.8.1
```
Cheers!

**Comment by o-felixz** [comment]
@Pitrified , based in @ffe4 's response, it would seem the issue is planned to be fixed in `2.8.2`, not `2.8.1` as you were told on SO. Seems like `2.8.2` has not yet been released though :cry:

**Comment by Pitrified** [comment]
Ah no worries, I'm not even the OP from that question, I just thought the fix was already live yet the error persisted, but if it's going to be fixed in the next release then no problem.
Cheers!
